pkg/kubectl/cmd/util/factory.go

```diff
@@ -303,6 +307,34 @@ func (f *Factory) BindFlags(flags *pflag.FlagSet) {
 	flags.SetNormalizeFunc(util.WordSepNormalizeFunc)
 }
 
+func (f *Factory) AddDeprecatedFlagShorthands(shorthand string) error {
```

Comment:
How about moving this code into `pflag`?  It looks like this would be a good complementary method to https://github.com/kubernetes/kubernetes/blob/master/Godeps/_workspace/src/github.com/spf13/pflag/flag.go#L283 and detection you wrote below would function well inside https://github.com/kubernetes/kubernetes/blob/master/Godeps/_workspace/src/github.com/spf13/pflag/flag.go#L299 against the pre-normalized name.  The details (`flag.Deprecated`) could be used to explain that only `-t` is gone.

We have a couple committers on pflags.  @eparis Any objections to that approach instead of adding this here?